### PR TITLE
Configure s2s secrets for functional tests

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-683

### Change description ###
Configure s2s secrets which will be used in functional tests.
Added a microservice key for the tests in hmcts/service-auth-provider-app#160

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: infrastructure/output.tf [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
 }
+
+output "TEST_S2S_URL" {
+  value = "${local.s2s_url}"
+}
+
+output "TEST_S2S_NAME" {
+  sensitive = true
+  value     = "${var.test_s2s_name}"
+}
```
Comment (rkondratowicz) [inline infrastructure/output.tf]: missing secret?

File: infrastructure/test_config.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  key_vault_id = "${data.azurerm_key_vault.key_vault.id}"
+  name         = "test-s2s-secret"
+  value        = "${data.azurerm_key_vault_secret.source_test_s2s_secret.value}"
+}
+
+resource "azurerm_key_vault_secret" "s2s_secret" {
+  key_vault_id = "${data.azurerm_key_vault.key_vault.id}"
+  name         = "s2s-secret"
+  value        = "${data.azurerm_key_vault_secret.s2s_secret.value}"
+}
```
Comment (rkondratowicz) [inline infrastructure/test_config.tf]: is this related to tests?

Comment (aliveni) [reply infrastructure/test_config.tf]: no, removed it

File: infrastructure/main.tf [hunk 3]
```diff
@@ -7,6 +7,10 @@ locals {
   s2s_rg              = "rpe-service-auth-provider-${local.local_env}"
   s2s_url             = "http://${local.s2s_rg}.service.core-compute-${local.local_env}.internal"
   ccd_api_url         = "http://ccd-data-store-api-${local.local_env}.service.core-compute-${local.local_env}.internal"
+
+  previewVaultName    = "${var.product}-aat"
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: preview is not needed ?

Comment (aliveni) [reply infrastructure/main.tf]: removed it now. 
I'll test the pr on sandbox

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -33,4 +46,7 @@ withPipeline(type, product, component) {
   enableDockerBuild()
   installCharts()
   enableSlackNotifications(channel)
+  loadVaultSecrets(secrets)
+
+  setVaultName('bulk-scan')
```
Comment (nitinprabhu) [inline Jenkinsfile_CNP]: should this be not `bulk-scan-sample-app` ?
